app/extensions/brave/content/scripts/metaScraper.js

```diff
@@ -0,0 +1,102 @@
+      wrap(html => requestHandlerApi.getValue(html.querySelectorAll('#content img[src]'), requestHandlerApi.getSrc)),
+      wrap(html => requestHandlerApi.getSrc(html.querySelector('img[alt*="author"]'))),
+      wrap(html => requestHandlerApi.getSrc(html.querySelector('img[src]')))
+    ]
+  },
+
+  getTitleRules: () => {
+    const wrap = rule => ({htmlDom}) => {
+      const value = rule(htmlDom)
+      return requestHandlerApi.isString(value) && requestHandlerApi.titleize(value)
+    }
+
+    return [
+      // Regular
+      wrap(html => requestHandlerApi.getContent(html.querySelector('meta[property="og:title"]'))),
+      wrap(html => requestHandlerApi.getContent(html.querySelector('meta[name="twitter:title"]'))),
+      wrap(html => requestHandlerApi.getContent(html.querySelector('meta[name="sailthru.title"]'))),
+      wrap(html => requestHandlerApi.getText(html.querySelector('.post-title'))),
+      wrap(html => requestHandlerApi.getText(html.querySelector('.entry-title'))),
+      wrap(html => requestHandlerApi.getText(html.querySelector('[itemtype="http://schema.org/BlogPosting"] [itemprop="name"]'))),
+      wrap(html => requestHandlerApi.getText(html.querySelector('h1[class*="title"] a'))),
+      wrap(html => requestHandlerApi.getText(html.querySelector('h1[class*="title"]'))),
+      wrap(html => requestHandlerApi.getText(html.querySelector('title')))
+    ]
+  },
+
+  getAuthorRules: () => {
+    const wrap = rule => ({htmlDom}) => {
+      const value = rule(htmlDom)
+
+      return requestHandlerApi.isString(value) &&
+        !requestHandlerApi.isUrl(value, {relative: false}) &&
+        requestHandlerApi.titleize(value, {removeBy: true})
+    }
+
+    return [
+      // Youtube
+      wrap(html => requestHandlerApi.getText(html.querySelector('#owner-name'))),
+      wrap(html => requestHandlerApi.getText(html.querySelector('#channel-title'))),
+      wrap(html => requestHandlerApi.getValue(html.querySelectorAll('[class*="user-info"]'))),
+      // Regular
+      wrap(html => requestHandlerApi.getContent(html.querySelector('meta[property="author"]'))),
+      wrap(html => requestHandlerApi.getContent(html.querySelector('meta[property="article:author"]'))),
+      wrap(html => requestHandlerApi.getContent(html.querySelector('meta[name="author"]'))),
+      wrap(html => requestHandlerApi.getContent(html.querySelector('meta[name="sailthru.author"]'))),
+      wrap(html => requestHandlerApi.getValue(html.querySelectorAll('[rel="author"]'))),
+      wrap(html => requestHandlerApi.getValue(html.querySelectorAll('[itemprop*="author"] [itemprop="name"]'))),
+      wrap(html => requestHandlerApi.getValue(html.querySelectorAll('[itemprop*="author"]'))),
+      wrap(html => requestHandlerApi.getContent(html.querySelector('meta[property="book:author"]'))),
+      requestHandlerApi.strict(wrap(html => requestHandlerApi.getValue(html.querySelectorAll('a[class*="author"]')))),
```

Comment:
`requestHandlerApi.strict` seems to take a string as input but `wrap` returns a function, so i'm not sure how this could work?

Reply:
if this is supposed to be checking that the final value is a "restricted" string, it would be clearer to just add a `isStrict` boolean option to `wrap`. https://github.com/brave/browser-laptop/pull/13115/files#diff-db4e6972ff167826cad220e8465ff2acR24

Reply:
I tried to mimic scraper code as much as possible, so that when doing upgrades we will not have a lot of problems, code changes

Reply:
discussed in person; `requestHandlerApi.strict` will be changed to mimic https://github.com/microlinkhq/metascraper/blob/master/packages/metascraper-author/index.js#L30 so that the function signature will be the same as `wrap`

Reply:
done